File: KKuTu-Web/src/main/java/com/horyu1234/kkutuweb/CharResponseWrapper.java
package com.horyu1234.kkutuweb;

import javax.servlet.http.HttpServletResponse;
import javax.servlet.http.HttpServletResponseWrapper;
import java.io.CharArrayWriter;
import java.io.PrintWriter;

/**
 * Created by horyu on 2018-01-15
 */
public class CharResponseWrapper extends HttpServletResponseWrapper {
    private final CharArrayWriter output = new CharArrayWriter();

    @Override
    public String toString() {
        return output.toString();
    }

    CharResponseWrapper(HttpServletResponse response) {
        super(response);
    }

    @Override
    public PrintWriter getWriter() {
        return new PrintWriter(output);
    }
}

File: kkutu-web/src/main/kotlin/com/horyu1234/kkutuweb/CharResponseWrapper.kt
package com.horyu1234.kkutuweb

import java.io.CharArrayWriter
import java.io.PrintWriter
import javax.servlet.http.HttpServletResponse
import javax.servlet.http.HttpServletResponseWrapper

class CharResponseWrapper internal constructor(response: HttpServletResponse) : HttpServletResponseWrapper(response) {
    private val output = CharArrayWriter()

    override fun toString(): String {
        return output.toString()
    }

    override fun getWriter(): PrintWriter {
        return PrintWriter(output)
    }
}